app/templates/pwa/ts/custom-service-worker.ts

```diff
@@ -0,0 +1,12 @@
+/*
```

Comment:
From what I'm reading, `webworker` and `dom` libraries are incompatible.

I'm trying to go deeper on this, as right now it would allow you to use `window` and `document` in service worker context, which it should not

Reply:
Found a resource explaining that the cleanest way is to use a separated tsconfig for src-pwa folder (TS resolves the config to use in the first tsconfig.json it finds going upward)

Reply:
Actually, I've tried creating a separate `tsconfig.json` before. But, for whatever reason VSCode didn't like that. It behaves as the declared `self` is the type of `any`, thus throwing lots of errors. If I remove the declaration and hope that it would pick up correctly, then IntelliSense for web worker-related stuff stops working, which probably means VSCode doesn't pick up the `tsconfig.json` correctly.

Later I've noticed that it wasn't just VSCode, the compilation ends with errors such as:
> Cannot find name 'ServiceWorkerGlobalScope'
> Cannot find name 'PushEvent'
> Cannot find name 'WindowClient'
> ...

TBH I don't have any idea why this isn't working, so I've decided to go with adding a reference directive, it also helps when using JS with IntelliSense, which is kind of a bonus(_I was originally using this solution before using TS_).

Reply:
Pretty strange, I just tested it and seems to work without problems with VSCode... May you test again when you have time?
I tested it on a clean project. Maybe you're using an old version of q/app with an old TS version?